### PR TITLE
build/ops: create and use local state dir in Travis CI

### PR DESCRIPTION
**Pull request by smithfarm**
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by tampakrap** [comment]
it still says 'OUCH!!! DOCHAZKA_STATE_DIR {HOME}/state is not readable by me!

**Comment by smithfarm** [comment]
> it still says 'OUCH!!! DOCHAZKA_STATE_DIR {HOME}/state is not readable by me!

Yes, fixing that now.

**Comment by tampakrap** [comment]
ah not ready yet sorry

**Comment by smithfarm** [comment]
Yeah, I didn't have the quoting right in this line:

```
sudo sh -c "echo \"set( 'DOCHAZKA_STATE_DIR', \"$ENV{HOME}/state\" );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
```

and as a result `$ENV` was getting evaluated by bash, which replaced it with the empty string.

**Comment by smithfarm** [comment]
And now the problem is that, in the latest code, employee objects have a "sync" property but Common hasn't been updated yet. On it.

**Comment by smithfarm** [comment]
The Travis CI will not pass until CPAN updates to `App::Dochazka::Common` 0.205 (I submitted it just now).

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ addons:
   postgresql: '9.4'
 before_script:
   - sudo mkdir /etc/dochazka-rest
+  - sudo mkdir /var/lib/dochazka-rest
+  - sudo chmod 777 /var/lib/dochazka-rest
```
Comment (tampakrap) [inline .travis.yml]: how about `chown $USER` instead?

**Comment by tampakrap** [comment]
tests still fail https://travis-ci.org/smithfarm/dochazka-rest/builds/159806646

**Comment by smithfarm** [comment]
test this please

**Comment by smithfarm** [comment]
This looks like an environmental issue in Travis CI. The BAILOUT is not coming from the REST code.

**Comment by smithfarm** [comment]
Actually, maybe I'm wrong about that.

**Comment by smithfarm** [comment]
I found the BAIL_OUT(0) call that is triggering the failure, and added a commit to expose more information on the cause.

**Comment by smithfarm** [comment]
The problem is/was that auditing was active, but the `postgresql.conf` file in Travis CI environment does not have the `dochazka.eid = -1` line.

**Comment by smithfarm** [comment]
```
All tests successful.
Files=43, Tests=10549, 51 wallclock secs ( 2.02 usr  0.30 sys + 44.52 cusr  3.32 csys = 50.16 CPU)
Result: PASS
```

**Pull request merged**